src/docker/include/callbacks.h

```diff
@@ -0,0 +1,43 @@
+#ifndef CALLBACKS_H
+#define CALLBACKS_H
+
+#include <algorithm>
+#include <functional>
+#include <string>
+#include "version.h"
+
+namespace RestClient {
```

Comment:
I think you greatly simplify this by using libaktualizr's HttpClient class that has all this stuff. e.g.:

 https://github.com/foundriesio/aktualizr-lite/pull/46/files#diff-d2f08e43f860c947307b0b54cbbdfe1bbea2d669ec67ee3635b5bc82569e96cd

Reply:
Also - one clever thing I've started copying from @mike-sul is the use of factories for things like this so that you can create nice mocks in the test class.

Reply:
@doanac what is all that? all that we need is this....

`HttpClient client("/var/run/docker.sock");
 std::string cmd = "http://localhost/containers/json?" + app;
 root_ =  client.get(cmd, HttpInterface::kNoLimit).getJson();
`

Reply:
btw if the requirement we need is to have an occasional http://localhost/containers/json?all request (instead of per-app), let me know and I will amend...is really straightforward